src/model/codeAssignments.model.ts

```diff
@@ -0,0 +1,38 @@
+import {
+  JoinColumn,
+  OneToOne,
+  Entity,
+  Column,
+  PrimaryGeneratedColumn,
+  CreateDateColumn,
+  UpdateDateColumn,
+  DeleteDateColumn,
+} from 'typeorm'
+
+import AssignmentModel from './assignments.model'
+
+@Entity('code_assignment')
```

Comment:
Will do. I'm still not thrilled about the way we have this setup and I've been dreaming up ways to refactor the way we think of assignment types. We have three types, code, non code, and manual, but this is pretty restrictive. I'm thinking of having a separate submission type and grading type. So we could say either form submission, file submission, git submission them specify the grader as autograded through Leviathan, autograded locally, manually graded. Then we can mix and match all we want. I can picture scenarios where I'd want a form submission with an autograded when whatever we build for local grading just doesn't do what I want

Reply:
It's possible that these just belong as options under the assignment table and store it as json? Might be a bit more dynamic.

You've gotta be careful with json in sql cause it can go off the rails pretty fast. But so long as it's not super nesty, it shouldn't be an issue. It'll be more dynamic too which is probably better than having to add a dedicated table for every new assignment type

Reply:
I'd be more comfortable with that. If we have 2 Jason string, one for submission method and one for grading method, that would keep us flexible (dare I say, more agile). Adding a new table when someone wants a manually grader custom form or attendance with an autograded (I dunno, individualized attendance codes?) would be painful. Give then end user max flexibility and JSON let's us avoid messing with the db too much

Reply:
Well I still think it should be set in TS what kind of stuff the options would be used for, I don't want users to be able to submit whatever they want.

What is this very very specifically used for? We can give assignment creators options to build what they want to build, but it should be within defined properties I'd think. Assuming this property is being used anywhere within our code. If it's being used in their code instead, then it shouldn't even make it's way to the DB, and just be included with their grading stuff imo.

Reply:
I believe we're on the same page. When we add a new feature that can be used through the JSON strings we update ts types, but not db tables. For something specific to their grader, it should either be 1. Not our concern or 2. Something so common that we send it to every grader